server/src/main/java/org/elasticsearch/search/internal/InternalSearchResponse.java

```diff
@@ -40,16 +43,24 @@ public InternalSearchResponse(SearchHits hits, InternalAggregations aggregations
 
     public InternalSearchResponse(StreamInput in) throws IOException {
         super(
-                new SearchHits(in),
-                in.readBoolean() ? InternalAggregations.readFrom(in) : null,
-                in.readBoolean() ? new Suggest(in) : null,
-                in.readBoolean(),
-                in.readOptionalBoolean(),
-                in.readOptionalWriteable(SearchProfileResults::new),
-                in.readVInt()
+            new SearchHits(in),
+            in.readBoolean() ? InternalAggregations.readFrom(in) : null,
+            in.readBoolean() ? new Suggest(in) : null,
+            in.readBoolean(),
+            in.readOptionalBoolean(),
+            readProfile(in),
+            in.readVInt()
         );
     }
 
+    private static SearchProfileResults readProfile(StreamInput in) throws IOException {
```

Comment:
Do we need this special handling if we already have BWC checks in `SearchProfileResults(StreamInput)`?

Reply:
I'm going to have to drop some breakpoints on this one. It sure doesn't look like we should need it. It'd be nice to be able to remove `Writeable` from `SearchProfileQueryPhaseResults`. I thought we did, but I'll be we do not.

Reply:
I've figured it out - you were right - I was doing the bwc logic twice. Actually using one way on read and one on write. Bad idea! I'm dropping this method entirely and letting the `SearchProfileResults` handle the read.